src/averageAges.js

```diff
@@ -20,24 +20,49 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+
+  let malesAverageAge = people.filter((person) => person.sex === 'm');
+
+  malesAverageAge = century !== undefined
+    ? malesAverageAge.filter(
+      (person) => Math.ceil(person.died / 100) === century
+    )
+    : malesAverageAge;
+
+  return malesAverageAge.reduce(
+    (accumulator, currentPerson) =>
+      accumulator + (currentPerson.died - currentPerson.born), 0
+  ) / malesAverageAge.length;
 }
 
 /**
  * Implement calculateWomenAverageAge function
  *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
+ * Function returns average age of women in array. If `withkids` is
+ * specified then function calculates average age only for women with kids
  *
- * Hint: To check if a woman has children you should find someone who mention
+ * Hint: To check if a woman has kids you should find someone who mention
  * her as mother.
  *
  * @param {object[]} people
- * @param {boolean} withChildren - optional
+ * @param {boolean} withkids - optional
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
+function calculateWomenAverageAge(people, withkids) {
   // write code here
+  let femalesAverageAge = people.filter((person) => person.sex === 'f');
+  const notMotherless = people.filter((person) => person.mother !== null);
+
+  femalesAverageAge = withkids === true
```

Comment:
same, you don't have to write it like `variable === true`, just `variable` will do the same **in this case**